tests/test_core.py

```diff
@@ -23,15 +24,112 @@ class test_PairMatrix(unittest.TestCase):
 
     def test_init(self):
         """Test construction with different list types."""
-        raise NotImplementedError()
+
+        pair_types = ('A','B')
+        pair_dict = {('A','B'):{}, ('B','B'):{}, ('A','A'):{}}
+
+        #test construction with tuple input
+        m = relentless.core.PairMatrix(types=('A','B'))
+        self.assertEqual(m.types, pair_types)
+        self.assertEqual(m._data, pair_dict)
+
+        #test construction with list input
+        m = relentless.core.PairMatrix(types=['A','B'])
+        self.assertEqual(m.types, pair_types)
+        self.assertEqual(m._data, pair_dict)
+
```

Comment:
Can we do at least one test where the type is not a string? Perhaps an int? This might be a little silly for the user to do, but the class should still work. You can test all ints, and then also mixes of ints and strings. You might want to add it as a separate test method.

Another test that would be good is to use a string int vs. a real int, which are technically different keys, i.e., `"1"` vs. `1`.